python/ray/serve/_private/common.py

```diff
@@ -28,6 +28,15 @@ class DeploymentID:
     name: str
     app_name: str = SERVE_DEFAULT_APP_NAME
 
+    def __repr__(self):
+        s = f"Deployment(name='{self.name}'"
+        if not self.app_name or self.app_name == SERVE_DEFAULT_APP_NAME:
```

Comment:
Non-blocker: I kinda feel it's fine to print "default" as the app name just in case if the user doesn't know what's the default value we use. Also, in the case when the user explicitly set it to be "default", it won't be omitted and cause confusion.

Reply:
Agreed, I also think it would be more clear if we print "default" as the app name. Also, now that 1.x deployments are fully deprecated, I think `app_name` should always be non-empty

Reply:
Ok, I will update it to include `app='default'`